### PR TITLE
[CDAP-16930] handling missing artifacts in studio and detail view

### PR DESCRIPTION
**Pull request by itsanudeep**
Cherry-pick of #12466 

Jira: https://issues.cask.co/browse/CDAP-16930

Plugin artifacts could be missing from the pipeline either in the studio or in the detail view and currently we do not communicate this to the user in any form.

This PR

- Adds an error badge to the node indicating there's something wrong with the node in studio.
- Disables the properties button with a tooltip in studio and detail view.
- If users try to deploy the pipeline with missing artifact, they would see an error banner.
- Fixes an issue where we incorrectly populate pluginsMap (Jira: https://issues.cask.co/browse/CDAP-17042)

### CODE REVIEW

**Review by elfenheart — DISMISSED** [review]

File: cdap-ui/app/directives/dag-plus/my-dag-ctrl.js [hunk 1]
```diff
@@ -1685,6 +1686,10 @@ angular.module(PKG.name + '.commons')
 
     let subAvailablePlugins = AvailablePluginsStore.subscribe(() => {
       vm.pluginsMap = AvailablePluginsStore.getState().plugins.pluginsMap;
+      $scope.nodes.forEach(node => {
+        let key = generatePluginMapKey(node);
```
Comment (yukiej) [inline cdap-ui/app/directives/dag-plus/my-dag-ctrl.js]: const?

File: cdap-ui/cypress/integration/pipeline.plugin.missing.spec.ts [hunk 2]
```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
```
Comment (yukiej) [inline cdap-ui/cypress/integration/pipeline.plugin.missing.spec.ts]: Nit: license year should be updated

**Review by yukiej — APPROVED** [review]
Two extremely minor nits, otherwise lgtm!

**Comment by elfenheart** [comment]
Build: https://builds.cask.co/browse/CDAP-URUT339

**Pull request merged**